### DISCUSSION TITLE
cors header ‘access-control-allow-origin’ missing

### DISCUSSION THREAD

**Discussion by Bryan-u-2** [fastapi/fastapi#6959] [Questions]
![image](https://user-images.githubusercontent.com/66239183/104408479-1d476c80-558a-11eb-9f91-02cda5e656ce.png)
How do i fix this?

**Comment by krishnardt** [comment]
access-control-allow-origin helps us to identify if the resource used by the request is present in the current origin or not.
In order to avoid that...
we have to add the specific domain, if we have to access that"specific domain" api/resource.

If the access-control-allow-origin is missing in the request, we can add it by appending it in the request.
And if the server that is being accessed in your control, add it in the server.

refer following links for reference.
[https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSMissingAllowOrigin](url)
[https://stackoverflow.com/questions/5584923/a-cors-post-request-works-from-plain-javascript-but-why-not-with-jquery](url)
[https://fastapi.tiangolo.com/tutorial/cors/](url)

In fastapi, you can add using CORSMiddleware.


```
from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

origins = [
    "http://localhost.tiangolo.com",
    "https://localhost.tiangolo.com",
    "http://localhost",
    "http://localhost:8080",
]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)


@app.get("/")
async def main():
    return {"message": "Hello `World"}
```

**_allow_origins parameter is where you give access to the domains.
If you want to give access to all origins... replace origins = ['*']_**




You can also set the ACAO header from the webpage using javascript either by XMLhttpRequest or fetchapi.

**Comment by tiangolo** [comment]
I imagine you solved your problem, so thanks for coming back to close the issue @Bryan-u-2 ☕ 

If anyone else is having problems, the first step is to read the docs: https://fastapi.tiangolo.com/tutorial/cors/

And then create a new issue with a minimal, self-contained example to reproduce it.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
